### DISCUSSION TITLE
Deletion of data objects extremely slow

### DISCUSSION THREAD

**Discussion by jratzenboeck** [pimcore/pimcore#15449] [General]
Hello!

I am part of the development team of a large project where we have to manage a lot of different data objects with quite an amount of relations. Right now I am writing a command, where we can automatically delete multiple objects. Now, I am facing the issue that deletion of single objects is extremely slow if they have a bigger amount of dependencies (ie. the main products in my case). The fact that it might take a few seconds to delete these would be fine to me but it takes multiple minutes to delete just one object, which is far too long to me. 

I am rather new to Pimcore and don't know all the inner workings. Unfortunately I could not really find anything regarding this topic and thus I am asking here. Can anyone help me how I can implement that to make this process faster?

Thanks for your help in advance.

**Comment by kingjia90** [comment]
Which version of Pimcore is the project? Hopefully it's not 10.5.24, see also https://github.com/pimcore/pimcore/pull/15420#issuecomment-1602501798 in case and update to latest

**Reply by jratzenboeck** [reply]
No, we currently have version 10.5.16. Do you know if this only occurs in this one particular version or could it be that our version is also affected and we have to upgrade to > 10.5.25?

**Reply by kingjia90** [reply]
Ok, then it's not related to the known issue. 
Unfortunately, i am not aware of any other possible regression, but i will write it down if i find something or something come up to my mind
